integrations/remote-data-blocks.php

```
@@ -25,6 +25,38 @@ public function is_loaded(): bool {
 		return defined( 'REMOTE_DATA_BLOCKS__LOADED' );
 	}
 
+	/**
+	 * Returns the current WordPress version.
+	 *
+	 * Wraps the global `$wp_version` variable in a function to make it easier to mock in tests.
+	 */
+	public function get_wp_version(): string {
+		global $wp_version;
+		return $wp_version;
+	}
+
+	/**
+	 * Returns `true` if the current WordPress version is supported by Remote Data Blocks.
+	 *
+	 * @return bool `true` if the current WordPress version greater than or equal to the minimum
+	 *              WordPress version defined in the environment config or if no minimum WordPress
+	 *              version is defined, `false` otherwise.
+	 */
+	public function is_supported_wp_version(): bool {
+		$wp_version = $this->get_wp_version();
+
+		$config = $this->get_env_config();
+		if (
+			isset( $config['minimum_wp_version'] ) &&
+			is_string( $config['minimum_wp_version'] ) &&
```

Shouldn't this method return `false` if `$config['minimum_wp_version']` is not correctly set, instead of defaulting to `true`?

---

Made it return `true` to account for case where GOOP change was deployed after this or the configmap changes weren't propagated. Now that GOOP is deployed, I think we could default to being strict where if `minimum_wp_version` is not in config we don't load the plugin. 

Though `minimum_wp_version` acts like an additional filter, and wouldn't absence of a filter indicate that needn't be applied ?

---

> Though minimum_wp_version acts like an additional filter, and wouldn't absence of a filter indicate that needn't be applied ?

This is how I would look at it. If the value isn't there, don't apply the logic and let it load like normal.

---

I see that logic. But since this is designed to prevent fatals on customer sites, I was thinking more from a security / protective stance. If we expect this guard to always be in place, do we want it to "fail open" (permissive) or "fail closed" (restrictive)?

---

Ie, what happens if a bug in the code that provides this config accidentally stops providing `minimum_wp_version`? What do we want to happen?

---

Moved to skip loading RDB when `minimum_wp_version` is not avaiable